### ISSUE TITLE
[Linux] [Ubuntu 22.04] Assertion Failed

### ISSUE THREAD

**Issue by killerdevildog11** [xenia-project/xenia#2080]
### Validation

- [X] I've read the [FAQ](https://github.com/xenia-project/xenia/wiki/FAQ).
- [X] The Xenia build used is from the master branch (not MLBS/AlexVS/Canary/pull requests, etc.)
- [X] This issue isn't for tech support (help with Xenia).
- [X] If this issue occurs in a specific game, I've done analysis to locate the faulty subsystem of the emulator and a potential reason in it.
- [X] I've checked if this issue hasn't already been reported.
- [X] My device meets the minimum requirements: https://github.com/xenia-project/xenia/wiki/Quickstart#system-requirements
- [X] (If building) I have read the building doc: https://github.com/xenia-project/xenia/blob/master/docs/building.md

### Describe what's going wrong

When running the program and selecting file>showcontentdirectory the core dumps and says an assertion failed. 

### Describe what should happen

I am assuming it should open nautilus or whatever file explorer is on default

### If applicable, provide a callstack here, especially for crashes

    MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

    i> 000C22A2 Vulkan device: Intel(R) UHD Graphics (TGL GT1) (vendor 8086, device 9A60, driver 05800005, API 1.3.204)
    i> 000C22A2 Vulkan device extensions:
    i> 000C22A2 * VK_EXT_fragment_shader_interlock: yes
    i> 000C22A2   * Sample interlock: yes
    i> 000C22A2   * Pixel interlock: yes
    i> 000C22A2 * VK_EXT_memory_budget: yes
    i> 000C22A2 * VK_EXT_shader_demote_to_helper_invocation: yes
    i> 000C22A2   * Demote to helper invocation: yes
    i> 000C22A2 * VK_EXT_shader_stencil_export: yes
    i> 000C22A2 * VK_KHR_bind_memory2: yes
    i> 000C22A2 * VK_KHR_dedicated_allocation: yes
    i> 000C22A2 * VK_KHR_get_memory_requirements2: yes
    i> 000C22A2 * VK_KHR_image_format_list: yes
    i> 000C22A2 * VK_KHR_maintenance4: yes
    i> 000C22A2 * VK_KHR_portability_subset: no
    i> 000C22A2 * VK_KHR_sampler_ycbcr_conversion: yes
    i> 000C22A2 * VK_KHR_shader_float_controls: yes
    i> 000C22A2   * Signed zero, inf, nan preserve for float32: yes
    i> 000C22A2   * Denorm flush to zero for float32: yes
    i> 000C22A2 * VK_KHR_spirv_1_4: yes
    i> 000C22A2 * VK_KHR_swapchain: yes
    i> 000C22A2 Added handle:F8000004 for N2xe6kernel7XObjectE
    i> 000C22A2 XThreadF8000004 (1) Stack: 70010000-70030000
    i> 000C22A2 Added handle:F8000008 for N2xe6kernel7XObjectE
    K> F8000004 XThread::Execute thid 1 (handle=F8000004, 'GPU Commands (F8000004)', native=66E3E640, <host>)
    G> F8000004 Vulkan sparse binding is not used for shared memory emulation - video memory usage may increase significantly because a full 512 MB buffer will be created
    i> 000C22A2 XThreadF8000008 (2) Stack: 70050000-70070000
    K> F8000008 XThread::Execute thid 2 (handle=F8000008, 'GPU VSync (F8000008)', native=65E3D640, <host>)
    G> F8000004 VulkanTextureCache: Format k_2_10_10_10 (signed) is not supported by the device (preferred Vulkan format is 65)
    G> F8000004 VulkanTextureCache: Format k_2_10_10_10_AS_16_16_16_16 (signed) is not supported by the device (preferred Vulkan format is 65)
    ../src/intel/isl/isl.c:2220: FINISHME: ../src/intel/isl/isl.c:isl_surf_supports_ccs: CCS for 3D textures is disabled, but a workaround is available.
    i> 000C22A2 Added handle:F800000C for N2xe6kernel7XObjectE
    i> 000C22A2 XThreadF800000C (3) Stack: 70090000-700B0000
    i> 000C22A2 Added handle:F8000010 for N2xe6kernel7XObjectE
    K> F800000C XThread::Execute thid 3 (handle=F800000C, 'XMA Decoder (F800000C)', native=4FFFF640, <host>)
    i> 000C22A2 XThreadF8000010 (4) Stack: 700D0000-700F0000
    i> 000C22A2 Added handle:F8000014 for N2xe6kernel7XObjectE
    K> F8000010 XThread::Execute thid 4 (handle=F8000010, 'Audio Worker (F8000010)', native=4EFFE640, <host>)
    i> 000C22A2 Added handle:F8000018 for N2xe6kernel7XObjectE
    i> 000C22A2 Added handle:F800001C for N2xe6kernel7XObjectE
    i> 000C2299 VulkanPresenter: Created 1280x720 swapchain with format 44, color space 0, presentation mode 0
    xenia: ../src/xenia/base/system_gnulinux.cc:32: void xe::LaunchFileExplorer(const std::filesystem::path &): Assertion `false' failed.
    Aborted (core dumped)

### If applicable, upload a logfile and link it here

[xenia.log](https://github.com/xenia-project/xenia/files/9873386/xenia.log)

**Comment by i-nod** [comment]
change the function in src/xenia/base/system_gnulinux.cc called launchfileexplorer from the current stub to:
  void LaunchFileExplorer(const std::filesystem::path& path) {
    auto cmd = std::string("xdg-open ");
    cmd.append(path.c_str());
    system(cmd.c_str());
  }

and rebuild then it should function as desired.

**Comment by dmanlfc** [comment]
@i-nod that works, but sadly no video for any games :-(

**Comment by i-nod** [comment]
@dmanlfc yeah I know. I made fixes for the file system stuff too.  It needed to look more like windows for it to properly open my xbox images.  Still though, even after the threads start up just a black screen.
